src/middleware/cloudWatch.ts

```diff
@@ -102,6 +81,51 @@ export class CloudWatchMiddleware
       );
     }
   };
+
+  // version 2
+  private extractTestTypesV2(data: PDTHealthCertV2): string[] {
+    let testTypes: string[] = [];
+    const validTestTypes = ["art", "pcr", "ser"];
+    if (typeof data.type === "string") {
+      testTypes = [data.type];
+    } else if (Array.isArray(data.type)) {
+      testTypes = data.type;
+    }
+    for (let i = 0; i < testTypes.length; i += 1) {
+      testTypes[i] = testTypes[i].toLowerCase();
+      if (!validTestTypes.includes(testTypes[i])) {
+        testTypes[i] = `UNRECOGNISED: ${testTypes[i]}`;
+      }
+    }
+    return testTypes;
+  }
+
+  // version 1
+  private extractTestTypesV1(data: HealthCertDocument): string[] {
+    const testTypes: string[] = [];
+    const observations = data.fhirBundle?.entry as Observation[];
+    const observation = observations.find(
+      (entr: any) => entr.resourceType === "Observation"
+    ) as Observation;
+    let { display } = observation.code.coding[0]; // e.g. Reverse transcription polymerase chain reaction (rRT-PCR) test
```

Comment:
KIV: Can also match by `code` (e.g. `94531-1`) instead of `display`.

Perhaps we can see how well this string matching performs first.